MonoMod.RuntimeDetour/Detour.cs

```diff
@@ -385,10 +400,29 @@ public Detour(Expression<Action> from, Expression<Action> to)
                 return;
 
             // GetNativeStart to avoid repins and managed copies.
-            _TopDetour = new NativeDetour(Method.GetNativeStart(), TargetReal.GetNativeStart());
+            _TopDetour = new NativeDetour(Method.Pin().GetNativeStart(), TargetReal.Pin().GetNativeStart());
```

Comment:
Method and TargetReal get pinned here for the top detour only but don't get unpinned until Free. I'm a little bit blind and worried right now so I'll better ask than be sorry afterwards: Is it supposed to be this way or are they supposed to be unpinned in TopUndo and only TopUndo instead?

Reply:
To be honest, I'm not entirely sure. Would there be any major consequence to leaving methods pinned forever? I went ahead and added Unpin calls in TopUndo to be safe, though I have no idea what the correct thing to do would be.

Reply:
Theoretically it would mostly affect Mono (untestable right now) and DynamicMethods - which if I'm being honest, the more I look at it and look back at my previous review, the less I see them being accounted for. Welp, I'll need to figure out what's going on with them before continuing to review this PR 😅